### PR TITLE
Handling conflicting errors for existing emails

### PR DESCRIPTION
**Pull request by Icon1cc**
image

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1 +1,3 @@
-VITE_BACKEND_URL=http://localhost:8080/
+# VITE_BACKEND_URL=http://localhost:8080
+
+VITE_BACKEND_URL=https://admin-kosjenka-dev.vercel.app
```
Comment (DoStini) [inline .env.example:3]: Dont change this, use the local .env

File: src/components/AdminForm.tsx [hunk 2]
```diff
@@ -31,37 +39,37 @@ const AdminForm: React.FC = () => {
       </div>
       <div className="flex-grow flex items-center justify-center p-8">
         <div className="bg-custom-light-grey p-12 rounded-lg shadow-xl w-full max-w-2xl">
+          {/* Show error message when there is one */}
+          {errorMessage && (
+            <div className="mb-4 text-center p-2 text-red-600 bg-red-100 rounded">
+              {errorMessage}
+            </div>
```
Comment (DoStini) [inline src/components/AdminForm.tsx:46]: Extract this code to a component so we can use it everywhere

File: src/components/AdminForm.tsx [hunk 3]
```diff
@@ -7,20 +7,28 @@ const AdminForm: React.FC = () => {
...
     e.preventDefault();
+    setErrorMessage(""); // Clear any existing error messages
     adminService
       .create(email, password, isSuperAdmin)
       .then(() => {
         navigate("/admins");
       })
       .catch((err) => {
-        console.log(err);
+        // Check the error and set an appropriate message
```
Comment (DoStini) [inline src/components/AdminForm.tsx:23]: This comment is not very useful, the code is clear and simple enough to document itself

File: src/components/AdminForm.tsx [hunk 4]
```diff
@@ -7,20 +7,28 @@ const AdminForm: React.FC = () => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
   const [isSuperAdmin, setIsSuperAdmin] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(""); // State to hold the error message
```
Comment (DoStini) [inline src/components/AdminForm.tsx:10]: This comment is not very useful, the code is clear and simple enough to document itself

File: src/components/AdminForm.tsx [hunk 5]
```diff
@@ -31,37 +39,37 @@ const AdminForm: React.FC = () => {
...
-              >
+              <label htmlFor="email" className="text-lg font-semibold text-gray-700 block">
                 Email address
               </label>
               <input
                 type="email"
                 id="email"
                 value={email}
                 onChange={(e) => setEmail(e.target.value)}
-                className="text-custom-black mt-1 px-4 py-3 bg-white border border-slate-300 placeholder-slate-400 focus:outline-none focus:border-blue-600 focus:ring-1 focus:ring-blue-600 block w-full rounded-md text-lg"
```
Comment (DoStini) [inline src/components/AdminForm.tsx:47]: Why did you remove text black?

File: src/components/AdminForm.tsx [hunk 6]
```diff
@@ -31,37 +39,37 @@ const AdminForm: React.FC = () => {
...
-              >
+              <label htmlFor="password" className="text-lg font-semibold text-gray-700 block">
                 Password
               </label>
               <input
                 type="password"
                 id="password"
                 value={password}
                 onChange={(e) => setPassword(e.target.value)}
-                className="text-custom-black mt-1 px-4 py-3 bg-white border border-slate-300 placeholder-slate-400 focus:outline-none focus:border-blue-600 focus:ring-1 focus:ring-blue-600 block w-full rounded-md text-lg"
```
Comment (DoStini) [inline src/components/AdminForm.tsx:64]: Why did you remove text black?

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/components/AdminForm.tsx (hunk 4 above)
Comment (Icon1cc) [reply src/components/AdminForm.tsx:10]: done

File: src/components/AdminForm.tsx (hunk 3 above)
Comment (Icon1cc) [reply src/components/AdminForm.tsx:23]: Removed

File: src/components/AdminForm.tsx (hunk 5 above)
Comment (Icon1cc) [reply src/components/AdminForm.tsx:47]: added

File: src/components/Alert.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+
+interface AlertProps {
+  message: string;
+}
+
+const Alert: React.FC<AlertProps> = ({ message }) => {
```
Comment (DoStini) [inline src/components/Alert.tsx:7]:
```suggestion
const Alert = ({ message }) : AlertProps => {
```

File: src/components/AdminForm.tsx [hunk 8]
```diff
@@ -31,12 +40,11 @@ const AdminForm: React.FC = () => {
       </div>
       <div className="flex-grow flex items-center justify-center p-8">
         <div className="bg-custom-light-grey p-12 rounded-lg shadow-xl w-full max-w-2xl">
+          {/* Show error message when there is one */}
```
Comment (DoStini) [inline src/components/AdminForm.tsx:43]: Useless comment, the code is clear enough

**Review by DoStini — CHANGES_REQUESTED** [review]

File: src/components/Alert.tsx [hunk 9]
```diff
@@ -0,0 +1,14 @@
+interface AlertProps {
+  message: string;
+}
+
+// Alert component without React.FC, with return type annotated as JSX.Element
+const Alert = ({ message }: AlertProps): JSX.Element => {
```
Comment (DoStini) [inline src/components/Alert.tsx:6]:
```suggestion
const Alert = ({ message }: AlertProps) => {
```

File: .env.example [hunk 10]
```diff
@@ -1 +1,2 @@
-VITE_BACKEND_URL=http://localhost:8080/
+VITE_BACKEND_URL=http://localhost:8080
+
```
Comment (DoStini) [inline .env.example:2]:
```suggestion
VITE_BACKEND_URL=http://localhost:8080/
```

**Review by DoStini — CHANGES_REQUESTED** [review]

**Review by DoStini — APPROVED** [review]
